### ISSUE TITLE
Feature request/question: re-attaching to debugger of a pod

### ISSUE THREAD

**Issue by wimdetroyer** [GoogleCloudPlatform/cloud-code-intellij#2783]
I was wondering if it was possible to un-attach and re-attach to a debugger of a running pod. When skaffold is run with the plug-in in debug mode, a debugger is attached to for every port, however I cant seem to find a good way to un-attach and re-attach. Normally i'd just make a debug configuration for a host at a specified port, but only the host part is static for my pods. Any ideas on how to achieve this?

Ideally it'd be a feature like this: https://cloud.google.com/code/docs/vscode/debug-k8-pod

**Comment by etanshaul** [comment]
hi @wimdetroyer . thanks for the request!

We have been discussing items related to this, but I want to make sure I capture the intent and your workflow. It sounds like there are a couple feature requests in here:

1) The ability to exclude certain services from being debugged. I'm not sure if you were hinting at this, but is this something that would be useful to you? Currently we attach to every debuggable service defined in the Skaffold yaml, but I can imaging a way to, at configuration time, exclude services from being debugged.

2) Attaching and reattaching a debugger. What is the workflow you are hoping to enable by doing this? Are you hoping to disable a debugger to re-activate Skaffold's watch mode (and then re-enable debugging later once you are satisfied with the changes)? Or are you merely trying to disable breakpoints from being hit temporarily (a workaround would be to just silence the breakpoints on the attached debugger). Regardless of your intended workflow, it think we do need to make the debuggers have the ability to reattach to the same port after detaching.

Also what language are you using? Java?

**Comment by wimdetroyer** [comment]
We program in java.

1: 
It'd be nice to start with 'skaffold debug' and not have all debuggers start up, but I wouldn't mind if it were the case if afterwards I could easily turn them off, and back on again. I will explain what we had a bit further down.

2:
The workflow right now is a bit iffy because you have to have the intention from the get-go to debug, and you cant just run 'skaffold dev' and then in the middle of the process start debugging. This is one minor nuisance but we solve this by using JRebel, a tool which brings hot reloading in java, making skaffolds redeploy mechanism unneccessary. So this isn't that big of a deal.

To further illustrate where we came from and what we're missing:

 We used to develop using docker, and we created containers that always ran on certain ports, and debug ports. 
An example:

  someservice:
    restart: always
    image: some image
    environment:
      - JAVA_OPTS=${COMMON_JAVA_OPTS} -Xms512m -Xmx640m -Dspring.profiles.active=some profiles
    volumes:
- a volume
    networks:
      - default
    ports:
      - "8017:8080"  # 'normal' port
      - "3017:3080" # debug port

So if we started this container, we could simply hook into and out of a debugger when we please after having created a debug config in intellij:

![image](https://user-images.githubusercontent.com/58472100/85579700-64483180-b63b-11ea-922c-db07635cce73.png)

However, using kubernetes the host and debug port are assigned randomly. We already have a 'static' host by using ingress, however we cant just make a debug configuration for one of our pods because we don't have a static debug port. 

Does this sound a bit more clear now?

**Comment by etanshaul** [comment]
@wimdetroyer yes it really helps to see your workflow as we think through the best way to improve this experience. I'll report back with any updates or questions as they come up. 

The initial fix on our end is to make sure that when you detach a debugger when starting in `debug` mode, that you can successfully reattach it. I consider it a bug that this doesn't work - currently when you hit `stop`, and then `debug` on the debug panel it fails to reattach. Beyond this, a more robust method to toggle Skaffold's watch mode and attach and detach debuggers on demand would be a big improvement - this area needs a bit of careful thought on our end. How does this sound?

**Comment by wimdetroyer** [comment]
> I consider it a bug that this doesn't work - currently when you hit stop, and then debug on the debug panel it fails to reattach

Yes exactly! A fix for this would mean a world of difference!
